frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.stories.tsx

```diff
@@ -427,13 +427,21 @@ export const CardVisualizationsDarkTheme = {
   },
 };
 
-const EXPLICIT_SIZE_WAIT_TIME = 300;
 function ScrollDecorator(Story: StoryFn) {
+  const asyncCallback = createAsyncCallback();
+
   useEffect(() => {
-    setTimeout(() => {
-      document.querySelector("[data-testid=embed-frame]")?.scrollBy(0, 9999);
-    }, EXPLICIT_SIZE_WAIT_TIME);
-  }, []);
+    const scrollContainer = document.querySelector("[data-testid=embed-frame]");
+    const intervalId = setInterval(() => {
```

Comment:
previously, we only attempted to scroll once, but if the embed dashboard isn't ready for whatever reason, we won't scroll and will produce a wrong Loki snapshot.

This approaches will retry until we could scroll then capture the screenshot, making it more resilient to failures.